src/js/iterator.js

```diff
@@ -234,6 +244,7 @@ export default class Iterator {
     while (true) {
       let comparison = compare(position, this.currentNodePosition)
 
+      this.cacheNodePosition()
```

Comment:
Meh, this doesn't seem too tough to maintain and I think the explicit control makes sense in a data structure this critical.